drivers/wifi/siwx91x/siwx91x_wifi.c

```diff
@@ -173,6 +173,63 @@ static int siwx91x_ap_disable(const struct device *dev)
 	return ret;
 }
 
+static int siwx91x_ap_sta_disconnect(const struct device *dev, const uint8_t *mac_addr)
+{
+	ARG_UNUSED(dev);
+	sl_wifi_interface_t interface;
+	sl_mac_address_t mac = { };
+	int ret;
+
+	__ASSERT(mac_addr, "mac_addr cannot be NULL");
+
+	memcpy(mac.octet, mac_addr, ARRAY_SIZE(mac.octet));
+	interface = sl_wifi_get_default_interface();
+
+	ret = sl_wifi_disconnect_ap_client(interface, &mac, SL_WIFI_DEAUTH);
+	if (ret) {
+		LOG_ERR("Failed	to disconnect: 0x%x", ret);
+		return -EIO;
+	}
+
+	return ret;
+}
+
+static sl_status_t siwx91x_on_ap_sta_connect(sl_wifi_event_t event, void *data,
+					     uint32_t data_length, void *arg)
+{
+	ARG_UNUSED(event);
+	struct siwx91x_dev *sidev = arg;
+	struct wifi_ap_sta_info sta_info = { };
+
+	__ASSERT(data, "data cannot be NULL");
+	__ASSERT(arg, "arg cannot be NULL");
+
+	strncpy(sta_info.mac, (uint8_t *)data, data_length);
+	sta_info.mac_length = data_length;
+	sta_info.link_mode = WIFI_LINK_MODE_UNKNOWN;
+
+	wifi_mgmt_raise_ap_sta_connected_event(sidev->iface, &sta_info);
+
+	return SL_STATUS_OK;
+}
+
+static sl_status_t siwx91x_on_ap_sta_disconnect(sl_wifi_event_t event, void *data,
+						uint32_t data_length, void *arg)
+{
+	ARG_UNUSED(event);
+	struct siwx91x_dev *sidev = arg;
+	struct wifi_ap_sta_info sta_info = { };
+
+	__ASSERT(data, "data cannot be NULL");
+	__ASSERT(arg, "arg cannot be NULL");
+
+	memcpy(sta_info.mac, (uint8_t *)data, data_length);
```

Comment:
The original type of `data` (`void *`) looks correct. I don't believe it make sense to cast it.